pkg/collector/corechecks/sbom/processor.go

```diff
@@ -35,13 +37,16 @@ type processor struct {
 func newProcessor(workloadmetaStore workloadmeta.Store, sender aggregator.Sender, maxNbItem int, maxRetentionTime time.Duration) *processor {
 	return &processor{
 		queue: queue.NewQueue(maxNbItem, maxRetentionTime, func(entities []*model.SBOMEntity) {
-			sender.SBOM([]sbom.SBOMPayload{
-				{
-					Version:  1,
-					Source:   &sourceAgent,
-					Entities: entities,
-				},
+			encoded, err := proto.Marshal(&model.SBOMPayload{
+				Version:  1,
+				Source:   &sourceAgent,
+				Entities: entities,
 			})
+			if err != nil {
+				log.Errorf("Unable to encode message: %+v", err)
+			}
```

Comment:
Would it make sense for this to return early to avoid submitting (potentially) nil payload?

Reply:
Indeed ! Good catch !
Thanks you for this finding.
It should now be addressed by https://github.com/DataDog/datadog-agent/pull/16084/commits/8d97bc142d1c07e5d862dfe3b536a23e8ce399d0.